wasmtime-wast/src/wast.rs

```diff
@@ -95,469 +51,331 @@ impl WastContext {
+                        }
+                        Err(e) => e,
+                    };
                     println!(
-                        "{}:{}: TODO: Check the assert_uninstantiable message: {}",
-                        filename, line, message
+                        "{}\nTODO: Check the assert_malformed message: {}",
+                        context(span),
+                        message
                     );
+                    drop(err);
                 }
-                CommandKind::AssertUnlinkable { module, message } => {
-                    let _err = self.module(None, module).expect_err(&format!(
-                        "{}:{}: unlinkable module was successfully linked",
-                        filename, line
-                    ));
+                AssertUnlinkable {
+                    span,
+                    mut module,
+                    message,
+                } => {
+                    let bytes = module.encode().map_err(adjust_wast)?;
+                    let err = match self.module(None, &bytes) {
+                        Ok(()) => bail!("{}\nexpected module to fail to link", context(span)),
+                        Err(e) => e,
+                    };
                     println!(
-                        "{}:{}: TODO: Check the assert_unlinkable message: {}",
-                        filename, line, message
+                        "{}\nTODO: Check the assert_unlinkable message: {}",
+                        context(span),
+                        message
                     );
+                    drop(err);
                 }
+                AssertReturnFunc { .. } => panic!("need to implement assert_return_func"),
             }
         }
 
         Ok(())
     }
 
     /// Run a wast script from a file.
-    pub fn run_file(&mut self, path: &Path) -> Result<(), WastFileError> {
-        let filename = path.display().to_string();
-        let buffer = read_to_end(path).map_err(|e| WastFileError {
-            filename,
-            line: 0,
-            error: WastError::IO(e),
-        })?;
-        self.run_buffer(&path.display().to_string(), &buffer)
-    }
-}
-
-fn read_to_end(path: &Path) -> Result<Vec<u8>, io::Error> {
-    let mut buf: Vec<u8> = Vec::new();
-    let mut file = fs::File::open(path)?;
-    file.read_to_end(&mut buf)?;
-    Ok(buf)
-}
-
-/// Helper to convert wasmtime features to WABT features; would be nicer as Into<WabtFeatures> but
-/// wasmtime-jit does not have a wabt dependency
-fn convert_features(features: &Features) -> WabtFeatures {
-    let mut wabt_features = WabtFeatures::new();
-    if features.simd {
-        wabt_features.enable_simd()
-    }
-    if features.multi_value {
-        wabt_features.enable_multi_value()
-    }
-    if features.bulk_memory {
-        wabt_features.enable_bulk_memory()
-    }
-    if features.threads {
-        wabt_features.enable_threads()
```

Comment:
Does this new crate support simd, multi-value, threads, etc.? Do you expect to make them configurable, or have them on by default?

Reply:
It does indeed support them! There's still some [unsupported proposals](https://github.com/alexcrichton/wast/issues) but they shouldn't be hard to add.

For now the crate just has everything enabled by default, and it'll break as the upstream specs change.